django_project/cplus_api/utils/worker_analysis.py

```diff
@@ -342,31 +346,37 @@ def create_and_upload_output_layer(
         is_final_output: bool, group: str,
         output_meta: dict = None) -> OutputLayer:
     filename = os.path.basename(file_path)
-    cog_name = (
-        f"{os.path.basename(file_path).split('.')[0]}"
-        f"_COG."
-        f"{os.path.basename(file_path).split('.')[1]}"
-    )
-    cog_path = os.path.join(
-        os.path.dirname(file_path),
-        cog_name
-    )
-    subprocess.run(
-        f"gdal_translate -of COG -co COMPRESS=DEFLATE {file_path} {cog_path}",
-        shell=True
-    )
+    if get_layer_type(file_path) == 0:
+        cog_name = (
+            f"{os.path.basename(file_path).split('.')[0]}"
+            f"_COG."
+            f"{os.path.basename(file_path).split('.')[1]}"
+        )
+        final_output_path = os.path.join(
+            os.path.dirname(file_path),
+            cog_name
+        )
+        subprocess.run(
+            (
+                f"gdal_translate -of COG -co COMPRESS=DEFLATE"
+                f" {file_path} {final_output_path}"
+            ),
+            shell=True
+        )
+    else:
+        final_output_path = file_path
     output_layer = OutputLayer.objects.create(
         name=filename,
         created_on=timezone.now(),
         owner=scenario_task.submitted_by,
         layer_type=BaseLayer.LayerTypes.RASTER,
```

Comment:
@zamuzakki is there possibility that the output is not raster? I see that you're using get_layer_type(file_path) in above lines, perhaps we can also change this line to use get_layer_type?

Reply:
Yes @danangmassandy . When I added mask layer shapefile in settings, mask layer output is shapefile (not raster). I've updated it

```
{
    "final_output": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/cplus_scenario_output_52e0.tif"
    ],
    "weighted_activities": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/weighted_activities/Alien_Plant_Removal_1129_cleaned.tif",
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/weighted_activities/Alien_Plant_Removal_ea73.tif"
    ],
    "normalized_pathways": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/normalized_pathways/Alien_Plant_Removal_6995.tif",
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/normalized_pathways/Avoided_Grassland_Conversion_27d7.tif"
    ],
    "normalized_activities": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/normalized_activities/Alien_Plant_Removal_e25b.tif"
    ],
    "mask_layer": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/mask_layer/mask.shp"
    ],
    "activities": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/activities/Alien_Plant_Removal_0654.tif"
    ]
}
```

Reply:
@zamuzakki it seems the mask_layer is part of the input layer that we download before running the analysis. Please exclude this directory from the output.

Reply:
@danangmassandy I've updated it:
https://github.com/kartoza/cplus-api/pull/35/files#diff-465ccbd99fbb69a2027d5c32b4cbf690f75458225c5e715d594c9ae89645e69eR77